processing/src/main/java/io/druid/segment/DoubleDimensionMergerV9.java

```diff
@@ -103,19 +114,27 @@ public Double convertSegmentRowValuesToMergedRowValues(Double segmentRow, int se
   @Override
   public void processMergedRow(Double rowValues) throws IOException
   {
+    if (rowValues == null) {
+      nullRowsBitmap.add(rowCount);
+    }
     serializer.serialize(rowValues);
+    rowCount++;
   }
 
   @Override
   public void writeIndexes(List<IntBuffer> segmentRowNumConversions, Closer closer) throws IOException
   {
-    // double columns do not have indexes
+    this.nullValueBitmapWriter = IndexMergerV9.createNullRowsBitmapWriter(
+        ioPeon,
+        dimensionName,
+        nullRowsBitmap,
+        indexSpec
+    );
   }
 
   @Override
   public boolean canSkip()
   {
-    // a double column can never be all null
     return false;
```

Comment:
how about skipping this if it is all null

Reply:
we do not know upfront here if the column is all null so cannot skip it here. 
For dictionary encoding we know the dictionary upfront and we can decide to skip it here. 
would prefer to add that as an optimization in a separate PR.